crates/core/tedge_agent/src/agent.rs

```diff
@@ -171,9 +181,36 @@ impl Agent {
         runtime.spawn(software_update_builder).await?;
         runtime.spawn(converter_actor_builder).await?;
         runtime.spawn(health_actor).await?;
+        runtime.spawn(tedge_to_te_converter).await?;
 
         runtime.run_to_completion().await?;
 
         Ok(())
     }
 }
+
+pub fn create_tedge_to_te_converter(
+    mqtt_actor_builder: &mut MqttActorBuilder,
+) -> Result<ConvertingActorBuilder<TedgetoTeConverter, TopicFilter>, anyhow::Error> {
+    let tedge_to_te_converter = TedgetoTeConverter::new();
+    let subscriptions: TopicFilter = vec![
+        "tedge/measurements",
+        "tedge/measurements/+",
+        "tedge/events/+",
+        "tedge/events/+/+",
+        "tedge/alarms/+/+",
+        "tedge/alarms/+/+/+",
+        "tedge/health/+",
+        "tedge/health/+/+",
+    ]
+    .try_into()?;
+
+    // Tedge to Te converter
+    let mut tedge_converter_actor =
+        ConvertingActor::builder("TedgetoTeConverter", tedge_to_te_converter, subscriptions);
+
+    tedge_converter_actor.add_input(mqtt_actor_builder);
+    tedge_converter_actor.register_peer(NoConfig, mqtt_actor_builder.get_sender());
```

Comment:
See here on [how to connect a converter](https://github.com/thin-edge/thin-edge.io/blob/main/crates/core/tedge_actors/src/converter.rs#L226).

```suggestion
    tedge_converter_actor.add_input(mqtt_actor_builder);
    tedge_converter_actor.add_sink(mqtt_actor_builder);
```

Reply:
fixed